projects/js-packages/publicize-components/src/utils/share-status.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Normalizes the share status object.
+ *
+ * @param {import( './types' ).PostShareStatus} shareStatus - Share status object.
+ * @return {import( './types' ).PostShareStatus | undefined} - Normalized share status object.
+ */
+export function normalizeShareStatus( shareStatus ) {
+	if ( ! shareStatus || ! ( 'shares' in shareStatus ) || ! shareStatus.done ) {
+		return;
```

Comment:
we should return the value as is here

```suggestion
		return shareStatus;
```